pkg/controller/elasticsearch/remotecluster/remoteca/controller.go

```diff
@@ -0,0 +1,474 @@
+	span, _ := apm.StartSpan(ctx, "create_or_update_remote_ca", tracing.SpanTypeApp)
+	defer span.End()
+	results := &reconciler.Results{}
+
+	localClusterKey := k8s.ExtractNamespacedName(local)
+	remoteClusterKey := k8s.ExtractNamespacedName(remote)
+
+	// Add watches on the CA secret of the local cluster.
+	if err := addCertificatesAuthorityWatches(r, remoteClusterKey, localClusterKey); err != nil {
+		return results.WithError(err)
+	}
+
+	// Add watches on the CA secret of the remote cluster.
+	if err := addCertificatesAuthorityWatches(r, localClusterKey, remoteClusterKey); err != nil {
+		return results.WithError(err)
+	}
+
+	log.V(1).Info(
+		"Setting up remote CA",
+		"local_namespace", localClusterKey.Namespace,
+		"local_name", localClusterKey.Namespace,
+		"remote_namespace", remote.Namespace,
+		"remote_name", remote.Name,
+	)
+
+	// Check if local CA exists
+	localCA := &corev1.Secret{}
+	if err := r.Client.Get(transport.PublicCertsSecretRef(localClusterKey), localCA); err != nil {
+		if !errors.IsNotFound(err) {
+			return results.WithError(err)
+		}
+		results.WithResult(defaultRequeue)
+	}
+
+	if len(localCA.Data[certificates.CAFileName]) == 0 {
+		log.Info(
+			"Cannot find local CA cert",
+			"local_namespace", localClusterKey.Namespace,
+			"local_name", localClusterKey.Namespace,
+		)
+		r.recorder.Event(local, v1.EventTypeWarning, EventReasonLocalCaCertNotFound, caCertMissingError("local", localClusterKey))
+		// CA secrets are watched, we don't need to requeue.
+		// If CA is created later it will trigger a new reconciliation.
+		return nil
+	}
+
+	// Check if remote CA exists
+	remoteCA := &corev1.Secret{}
+	if err := r.Client.Get(transport.PublicCertsSecretRef(remoteClusterKey), remoteCA); err != nil {
+		if !errors.IsNotFound(err) {
+			return results.WithError(err)
+		}
+		results.WithResult(defaultRequeue)
+	}
+
+	if len(remoteCA.Data[certificates.CAFileName]) == 0 {
+		log.Info("Cannot find remote CA cert",
+			"remote_namespace", remote.Namespace,
+			"remote_name", remote.Name,
+		)
+		r.recorder.Event(local, corev1.EventTypeWarning, EventReasonRemoteCaCertMissing, caCertMissingError("remote", remoteClusterKey))
+		return nil
+	}
+
+	// Create local relationship
+	if err := reconcileRemoteCA(ctx, r.Client, local, remoteClusterKey, remoteCA.Data[certificates.CAFileName]); err != nil {
+		return results.WithError(err)
+	}
+
+	// Create remote relationship
+	if err := reconcileRemoteCA(ctx, r.Client, remote, localClusterKey, localCA.Data[certificates.CAFileName]); err != nil {
+		return results.WithError(err)
+	}
+
+	return nil
+}
+
+// reconcileRemoteCA copies certificates authorities across 2 clusters
+func reconcileRemoteCA(
+	ctx context.Context,
+	c k8s.Client,
+	owner *esv1.Elasticsearch,
+	remote types.NamespacedName,
+	remoteCA []byte,
+) error {
+	span, _ := apm.StartSpan(ctx, "reconcile_remote_ca", tracing.SpanTypeApp)
+	defer span.End()
+
+	// Define the desired remote CA object, it lives in the remote namespace.
```

Comment:
I'm a bit lost here: that's the CA of the local cluster that we copy into the remote namespace?